firebase-firestore/src/main/java/com/google/firebase/firestore/model/mutation/MutationBatch.java

```diff
@@ -76,19 +76,16 @@ public MutationBatch(
    * remote document.
    *
    * @param documentKey The key of the document to apply mutations to.
-   * @param maybeDoc The document to apply mutations to.
+   * @param document The document to apply mutations to.
    * @param batchResult The result of applying the MutationBatch to the backend.
    */
-  @Nullable
-  public MaybeDocument applyToRemoteDocument(
-      DocumentKey documentKey, @Nullable MaybeDocument maybeDoc, MutationBatchResult batchResult) {
-    if (maybeDoc != null) {
-      hardAssert(
-          maybeDoc.getKey().equals(documentKey),
-          "applyToRemoteDocument: key %s doesn't match maybeDoc key %s",
-          documentKey,
-          maybeDoc.getKey());
-    }
+  public void applyToRemoteDocument(
+      DocumentKey documentKey, MutableDocument document, MutationBatchResult batchResult) {
+    hardAssert(
+        document.getKey().equals(documentKey),
+        "applyToRemoteDocument: key %s doesn't match maybeDoc key %s",
```

Comment:
nit: fix this outdated mention of "maybeDoc" in this assertion message.

Reply:
Fixed